sdk/core/azure-core/test/ut/curl_options.cpp

```diff
@@ -0,0 +1,269 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "gtest/gtest.h"
+
+#include <azure/core/context.hpp>
+#include <azure/core/http/curl/curl.hpp>
+#include <azure/core/http/http.hpp>
+#include <azure/core/http/pipeline.hpp>
+#include <azure/core/http/policy.hpp>
+#include <azure/core/response.hpp>
+
+#include <string>
+#include <vector>
+
+namespace Azure { namespace Core { namespace Test {
+
+// Test are specific to the curl transport adapter
+#ifdef BUILD_CURL_HTTP_TRANSPORT_ADAPTER
```

Comment:
Move this check out of the file.  There is no need to compile curl_options if we are NOT building the HTTP_TRANSPORT_ADAPTER tests